examples/zoom-provider/server/zoomprovider.js

```diff
@@ -0,0 +1,116 @@
+const request = require('request')
+const BASE_URL = 'https://zoom.us/v2'
+
+/**
+ * an example of a custom provider module. It implements @uppy/companion's Provider interface
+ */
+class ZoomProvider {
+  constructor (options) {
+    this.authProvider = ZoomProvider.authProvider
+  }
+
+  static get authProvider () {
+    return 'zoom'
+  }
+
+  // /users/{userId}/recordings
+  list ({ token, query }, done) {
+    const path = '/users/me/recordings'
+    const options = {
+      url: `${BASE_URL}${path}`,
+      method: 'GET',
+      json: true,
+      headers: {
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    request(options, (err, resp, body) => {
+      if (err) {
+        console.log(err)
+        done(err)
+        return
+      }
+      done(null, this._adaptData(body))
+    })
+  }
+
+  download ({ id, token }, onData) {
+    const downloadUrl = id
+    const options = {
+      url: downloadUrl,
+      method: 'GET',
+      json: true,
+      headers: {
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    request(options, (err, resp, body) => {
+      if (err) {
+        console.log(err)
+        return
+      }
+      // how can I test this?
+      const url = body.links.download
+      request.get(url)
+        .on('data', (chunk) => onData(null, chunk))
+        .on('end', () => onData(null, null))
+        .on('error', (err) => console.log(err))
+    })
+  }
+
+  _adaptData (res) {
+    const nextPagePath = res && res.nextPagePath ? `/users/me/recordings&next_page_token=${res.nextPagePath}` : ''
+
+    const data = {
+      username: null,
+      items: [],
+      nextPagePath
+    }
+
+    if (res.meetings) {
+      res.meetings.forEach(meeting => {
+        meeting.recording_files.filter(record => record.file_type === 'MP4').forEach((record, index) => {
+          data.items.push({
+            isFolder: undefined,
+            icon: '',
+            name: `${meeting.topic}_${record.recording_start}${index > 0 ? `_pt_${index + 1}` : ''}`,
+            mimeType: 'video/mp4',
+            id: record.download_path,
+            thumbnail: '',
+            requestPath: record.download_url,
+            modifiedDate: record.recording_start,
+            size: record.file_size,
+            custom: {
+              recordTrueId: record.id
+            }
+          })
+        })
+      })
+    }
+    return data
+  }
+
+  logout ({ token }, done) {
```

Comment:
I noticed that the token expiry between uppy didn't seem to match token expiry on the custom provider side (ie the token expires with Zoom before the Uppy token expires). Is there a configuration somewhere that I should be setting this with Uppy?

Reply:
this is the behaviour in general. Uppy sets the token to expire in 24hrs, however if the Provider's token expires before then, the user will be prompted to login again.

Reply:
Hi again! So i seems that when the token is expired, the user isn't prompted to login again (so I need to manually delete the uppy zoom token), I'm trying to figure out why. As far as I can tell, the error handling looks the same as the other providers (except I've updated the error code to match Zoom's status code for expired token), so I'm a bit confused here. Does anything jump out at you why this might be?